File: app/src/test/java/com/nytimes/android/sample/StoreIntegrationTest.java
package com.nytimes.android.sample;

import com.nytimes.android.external.store3.base.impl.Store;
import com.nytimes.android.external.store3.base.impl.BarCode;
import com.nytimes.android.external.store3.base.impl.StoreBuilder;

import org.junit.Before;
import org.junit.Test;

import io.reactivex.Single;

import static junit.framework.Assert.assertEquals;

public class StoreIntegrationTest {

    private Store<String, BarCode> testStore;

    @Before
    public void setUp() throws Exception {
        testStore = StoreBuilder.<String>barcode()
                .fetcher(barCode -> Single.just("hello"))
                .open();
    }

    @Test
    public void testRepeatedGet() throws Exception {
        String first = testStore.get(BarCode.empty()).blockingGet();
        assertEquals(first, "hello");

    }
}

File: app/src/test/java/com/nytimes/android/sample/StoreIntegrationTest.kt
package com.nytimes.android.sample

import com.nytimes.android.external.store3.base.impl.BarCode
import com.nytimes.android.external.store3.base.impl.Store
import com.nytimes.android.external.store3.base.impl.StoreBuilder
import junit.framework.Assert.assertEquals
import kotlinx.coroutines.runBlocking
import org.junit.Test
import java.util.concurrent.atomic.AtomicInteger

class StoreIntegrationTest {

    private val atomicInt = AtomicInteger(0)

    private val store: Store<String, BarCode> = StoreBuilder.barcode<String>()
            .fetcher { atomicInt.incrementAndGet().toString() }
            .open()

    @Test
    fun testRepeatedGet() {
        val barcode = BarCode.empty()
        runBlocking {
            val first = store.fresh(barcode)
            val same = store.get(barcode)
            assertEquals(first, same)
        }
    }
}
